### PR TITLE
Change MySQL global variables query to globalVariable function

### PR DESCRIPTION
**Pull request by bohutang**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Deprecate the old regular style and use the new globalVariable method #11832

### CODE REVIEW

**Comment by BohuTANG** [comment]
CC @alexey-milovidov 
Please give a 'can be tested' lable, thanks.

File: src/Functions/globalVariable.cpp [hunk 1]
```diff
@@ -43,19 +47,39 @@ class FunctionGlobalVariable : public IFunction
     DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
     {
         if (!checkColumnConst<ColumnString>(arguments[0].column.get()))
-            throw Exception("Agrument of function " + getName() + " must be constant string", ErrorCodes::BAD_ARGUMENTS);
+            throw Exception("Argument of function " + getName() + " must be constant string", ErrorCodes::BAD_ARGUMENTS);
```
Comment (alexey-milovidov) [inline src/Functions/globalVariable.cpp:50]: :+1:

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Also please look at #11089
We can easily close it with this PR :)

**Comment by BohuTANG** [comment]
> Also please look at #11089
> We can easily close it with this PR :)

Done

**Comment by phuslu** [comment]
sorry for jumping this thread, is possible add mysql "connection_id()" fake function in this stage?
some mysql frontend tool (e.g. mycli) need "connection_id()" during connect.

**Comment by BohuTANG** [comment]
> sorry for jumping this thread, is possible add mysql "connection_id()" fake function in this stage?
> some mysql frontend tool (e.g. mycli) need "connection_id()" during connect.

This patch is about global variables, connection_id() is function.
AFAIK, connection_id()  is not necessary, what's the case?

**Comment by alexey-milovidov** [comment]
Flaky test is reported in https://github.com/ClickHouse/ClickHouse/issues/11815

**Pull request merged**